charts/k8s-monitoring/templates/agent_config/_loki.river.txt

```diff
@@ -1,26 +1,31 @@
 {{ define "agent.config.loki" }}
+{{- with .Values.externalServices.loki }}
 // Grafana Cloud Loki
 local.file "loki_host" {
   filename  = "/etc/grafana-agent-credentials/loki_host"
 }
-
+{{ if .basicAuth.username }}
 local.file "loki_username" {
   filename  = "/etc/grafana-agent-credentials/loki_username"
 }
-
+{{- end }}
+{{ if .basicAuth.password }}
 local.file "loki_password" {
   filename  = "/etc/grafana-agent-credentials/loki_password"
   is_secret = true
 }
+{{- end }}
 
 loki.write "grafana_cloud_loki" {
   endpoint {
-    url = nonsensitive(local.file.loki_host.content) + "{{ .Values.externalServices.loki.writeEndpoint }}"
-
+    url = nonsensitive(local.file.loki_host.content) + "{{ .writeEndpoint }}"
+{{ if .basicAuth }}
     basic_auth {
-      username = local.file.loki_username.content
-      password = local.file.loki_password.content
+      {{ if .basicAuth.username }}username = local.file.loki_username.content{{ end }}
+      {{ if .basicAuth.password }}password = local.file.loki_password.content{{ end }}
```

Comment:
Is it necessary to guard the username and password with individual conditionals or is it enough just to have the `if .basicAuth` check?

Reply:
If password is set, but username is not, I want only one to be created.